share/blob.go

```diff
@@ -79,8 +82,8 @@ func NewBlobFromProto(pb *v1.BlobProto) (*Blob, error) {
 	if pb.NamespaceVersion > NamespaceVersionMax {
 		return nil, errors.New("namespace version can not be greater than MaxNamespaceVersion")
 	}
-	if len(pb.Data) == 0 {
-		return nil, errors.New("blob data can not be empty")
+	if pb.ShareVersion > MaxShareVersion {
```

Comment:
I think we could also check against `SupportedShareVersions`?

Reply:
If we wanted to do that we should add it in `NewBlob` rather than here